granary/bluesky.py

```diff
@@ -44,59 +53,53 @@ def _maybe_load_lexicons():
 def url_to_did_web(url):
   """Converts a URL to a did:web.
 
+  In AT Proto, only hostname-based web DIDs are supported.
+  Paths are not supported, and will be discarded.
+
   Examples:
   * 'https://foo.com' => 'did:web:foo.com'
-  * 'https://foo.com:3000' => 'did:web:foo.com%3A3000'
-  * 'https://bar.com/baz/baj' => 'did:web:bar.com:baz:baj'
-
-  https://w3c-ccg.github.io/did-method-web/#example-creating-the-did
+  * 'https://foo.com:3000' => 'did:web:foo.com'
+  * 'https://foo.bar.com/baz/baj' => 'did:web:foo.bar.com'
 
-  TODO: require https?
+  https://atproto.com/specs/did
 
   Args:
     url: str
 
   Returns: str
   """
   parsed = urllib.parse.urlparse(url)
-  if not parsed.netloc:
+  if not parsed.hostname:
     raise ValueError(f'Invalid URL: {url}')
 
-  did = f'did:web:{urllib.parse.quote(parsed.netloc)}'
-  if parsed.path:
-    did += f'{parsed.path.replace("/", ":")}'
-
-  return did.strip(':')
+  return f'did:web:{parsed.hostname}'
```

Comment:
if there's an explicit port, this currently silently drops it. should we fail instead? or log a warning and mention that behavior in the docstring? and maybe also log if there's a path? or fail on that too? especially since we fail on both path and port in `did_web_to_url`, so it's not currently symmetric.

Reply:
If we throw I think that will break the test data tests in a way that would be quite tricky to unpick - logging a warning seems easier to me if it's OK with you

Reply:
Sure, warning log and mention in docstring sgtm. I expect `did:web`s won't actually be used much or at all for the foreseeable future, in either Bridgy classic or Bridgy Fed.